resources/src/asset.rs

```diff
@@ -31,11 +73,93 @@ impl AssetIdentifier {
     pub fn asset_issuer<'a>(&'a self) -> &'a str {
         &self.asset_issuer
     }
+
+    /// A new Asset can be a native stellar, or a fully identified asset
+    pub fn new(
+        asset_type: String,
+        asset_code: Option<String>,
+        asset_issuer: Option<String>,
+    ) -> Result<AssetIdentifier, String> {
+        if asset_type == "native".to_string() {
+            Ok(AssetIdentifier {
+                asset_type: asset_type,
+                asset_code: "XLM".to_string(),
+                asset_issuer: "Stellar Foundation".to_string(),
+            })
+        } else {
+            if asset_code.is_none() || asset_issuer.is_none() {
+                Err("Code and issuer are required for non-native assets".to_owned())
```

Comment:
String errs are less ideal, we should implement an enum for this or return a unit `Err(())` since we map away the error anyhow.

This is the equivalent of `raise "some string"` in ruby and is frowned upon.

Reply:
Lets address this in a separate pr :)